data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -226,10 +226,16 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
 
     // if there are fewer deletes than a reasonable number to keep in memory, use a set
     if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
-      return Deletes.filter(records, this::pos, Deletes.toPositionIndex(filePath, deletes));
+      PositionDeleteIndex positionIndex = Deletes.toPositionIndex(filePath, deletes);
```

Comment:
Extract this to a separate variable on top, and use it in outside the else block?

Reply:
Not sure I got you, but `positionIndex` is used only in this block.

Reply:
Yea sorry you can ignore this one, I mis-read it.